### PR TITLE
Update Parser.php

### PR DESCRIPTION
**Pull request by ab02111996**
I tested this code on my JSON files and it works uncorrect with non-latin characters. The thing is standart PHP function strlen() returns wrong result when input string has russian letters. So I wrote my own function getLengthOfString() that get every letter in input string and returns count of letters.
PHP Version - 7.1

### CODE REVIEW

File: src/Parser.php [hunk 1]
```diff
@@ -1,13 +1,9 @@
 <?php
-
```
Comment (soyuka) [inline src/Parser.php]: can you revert the line changes, it makes the diff incorrect

File: src/Parser.php [hunk 2]
```diff
@@ -119,65 +97,58 @@ public function __construct($stream, ListenerInterface $listener, string $lineEn
...
     {
         $this->lineNumber = 1;
         $this->charNumber = 1;
         $eof = false;
-
         while (!feof($this->stream) && !$eof) {
             $pos = ftell($this->stream);
             $line = stream_get_line($this->stream, $this->bufferSize, $this->lineEnding);
-
+            $len = $this->getLengthOfString($line);
```
Comment (soyuka) [inline src/Parser.php]: IMO https://secure.php.net/manual/en/function.mb-strlen.php would be even more appropriate

File: src/Parser.php (hunk 1 above)
Comment (gonzofy) [reply src/Parser.php]: @ab02111996 Agree with this. Can't really merge as-is.

File: src/Parser.php [hunk 3]
```diff
@@ -609,18 +535,24 @@ private function endSpecialValue($value, string $stringValue): void
...
-
     /**
      * @throws ParsingException
      */
     private function throwParseError(string $message): void
     {
         throw new ParsingException($this->lineNumber, $this->charNumber, $message);
     }
+    
+    private function getLengthOfString(string $s) {
```
Comment (MAXakaWIZARD) [inline src/Parser.php]: why can't we just use `mb_strlen`?

**Pull request closed without merging**